book/src/qobject/generated-qobject.md

```diff
@@ -0,0 +1,111 @@
+Therefore implementing `Default` on the Rust struct is currently mandatory.
+
+The C++ object will defer any property state to the Rust struct, and is therefore only a thin wrapper.
+The data for `Q_PROPERTY`s is stored in the Rust struct. The property getters and setters modify the Rust struct internally.
+
+Additionally, CXX-Qt generates methods on this struct that help you interact with Qt functionality.
+This includes access to the [`CxxQtThread` struct](./cxxqtthread.md), as well as a function to emit signals.
+
+## Referencing another QObject
+
+The `qobject::T` type is defined both within the CXX-Qt bridge, as well as the surrounding module.
+The simplest way to reference it is from the surrounding module.
+
+Example:
+``` rust,ignore,noplayground
+// In file qt_types.rs
+#[cxx_qt::bridge]
+mod ffi {
+  #[cxx_qt::qobject]
+  #[derive(Default)]
+  pub struct MyObject {}
+}
+```
+
+``` rust,ignore,noplayground
+// In another module
+fn my_function(parameter: &crate::qt_types::qobject::MyObject) { /* ... */ }
+```
+
+Unfortunately, nested QObjects aren't yet supported by CXX-Qt.
+Progress can be tracked in [#299](https://github.com/KDAB/cxx-qt/issues/299).
+
+## `impl qobject::T`
+As mentioned before, the C++ QObject is exposed to Rust as an opaque CXX type.
+This allows you to implement methods for it in Rust using normal `impl` blocks.
+Because `qobject::T` is an opaque C++ type, the same rules as with normal CXX opaque types apply.
+Most importantly this means that the type may never be accessed by value or by mutable self reference directly.
+Rather, all methods must use either `&self` or `self: Pin<&mut Self>` as the self types. This prevents Rust from moving the data in memory, which would invalidate C++ pointers to it.
+
+For more information about pinning, refer to the [pin documentation](https://doc.rust-lang.org/std/pin/). 
+
+In addition to methods that extend the Rust interface of a `qobject::T`, you may also mark methods within an `impl qobject::T` block with the `#[qinvokable]` attribute.
+These methods will be exposed to the C++ QObject and can be called by C++ or QML.
+See the [QObject page](./qobject_struct.md#invokables) for more details.
+
+## Available Functions
+
+Every `qobject::T` struct provides the following methods:
+
+### Access to the Qt thread
+``` rust,ignore,noplayground
+fn qt_thread(&self) -> UniquePtr<CxxQtThread>
+```
+This function provides you with a handle to the Qt thread that the QObject resides in.
+This is helpful as the QObject itself does not implement [Send](https://doc.rust-lang.org/std/marker/trait.Send.html) nor [Sync](https://doc.rust-lang.org/std/marker/trait.Sync.html).
+The CxxQtThread however is Send and can therefore be moved into a different thread.
+By using the `queue` function on the CxxQtThread, you may then queue a Rust lambda onto the Qt thread again.
```

Comment:
```suggestion
By using the CxxQtThread's `queue` method, you may then queue a Rust lambda onto the Qt thread again.
```
This clarifies that it's not a free function that takes a `CxxQtThread` as a parameter.

Reply:
done